pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -16,6 +16,200 @@ limitations under the License.
 
 package policies
 
+// In order to decide whether a certificate should be reissued or not, we want
+// to be gathering the "state of the world" regarding that particular
+// certificate, which is the entire purpose of DataForCertificate. Along with
+// the certificate's secret, DataForCertificate also returns two separate
+// certificate requests: the "current" and the "next" certificate request.
+//
+// To understand the roles of the "current" and "next" certificate requests, let
+// us look at three different scenarios: A, B and C.
+//
+// DIAGRAM (A1): the first scenario is a certificate that just got created.
+// Since the certificate is brand new, only the "next" certificate request
+// exists. You can see below that the first revision is "1":
```

Comment:
Fwiw, the use of `Certificate` in all of this is incredibly confusing.

I usually think of `Certificate` as a thing issued by a CA.

Whereas, in this sentence, it's more of a template for a certificate, or perhaps an unsigned certificate?

Reply:
That's a valid point, we do tend to use the word "certificate" in many different contexts:

1. sometimes we use the word "certificate" to refer to a X.509 certificate (a blob of ASN.1-encoded data and then PEM-formated);
2. and sometimes we refer to "certificate" as one item of the Kubernetes custom resource `/apis/cert-manager.io/v1/certificates`.

In the above context, I am refering to "certificate" as (2) (a Kubernetes object). And you are right, it can be thought of as a "certificate template".

Not sure how to rephrase the above... should I, for example, make sure to specific which type of certificate I am refering to?

```go
// DIAGRAM (A1): in the first scenario, a user just created a Kubernetes
// Certificate CR (custom resource). Since the Certificate object is new, only
// the "next" certificate request exists. You can see below that the first
// revision is "1":
```

Does it help?

Reply:
Fixed in 8b41ec1d5, let me know if my phrasing still lacks precision 👍